packages/loader/container-loader/src/createAndLoadContainerUtils.ts

```diff
@@ -239,3 +279,131 @@ export async function loadFrozenContainerFromPendingState(
 		documentServiceFactory: new FrozenDocumentServiceFactory(),
 	});
 }
+
+/**
+ * Loads a summarizer container with the required headers, triggers an on-demand summary, and then closes it.
+ * Returns success/failure and an optional error for host-side handling.
+ *
+ * @legacy @beta
+ */
+export async function loadSummarizerContainerAndMakeSummary(
+	loadExistingContainerProps: ILoadExistingContainerProps,
+): Promise<LoadSummarizerSummaryResult> {
+	const { logger, configProvider, request: originalRequest } = loadExistingContainerProps;
+	const telemetryProps = {
+		loaderId: uuid(),
+		loaderVersion: pkgVersion,
+	};
+
+	const subMc = mixinMonitoringContext(
+		DebugLogger.mixinDebugLogger("fluid:telemetry", logger, {
+			all: telemetryProps,
+		}),
+		sessionStorageConfigProvider.value,
+		configProvider,
+	);
+	const mc = createChildMonitoringContext({
+		logger: subMc.logger,
+		namespace: "SummarizerOnDemand",
+	});
+	const loader = new Loader(loadExistingContainerProps);
+	const baseHeaders = originalRequest.headers;
+	const request = {
+		...originalRequest,
+		headers: {
+			...baseHeaders,
+			[LoaderHeader.cache]: false,
+			[LoaderHeader.clientDetails]: {
+				capabilities: { interactive: false },
+				type: "summarizer",
+			},
+			[DriverHeader.summarizingClient]: true,
+			[LoaderHeader.reconnect]: false,
+		},
+	};
+
+	const container = await loader.resolve(request);
+
+	mc.logger.send({
```

Comment:
i would get ride of a bunch of the logging calls, and just wrap the whole this in a performance event, that will log start and end consistently with our other logs, and provide timing information